packages/console/src/ConsoleObjectsMenu.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { ReactElement, useMemo, useRef, useState } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  Button,
+  DropdownActions,
+  DropdownMenu,
+  SearchInput,
+} from '@deephaven/components';
+import { vsGraph, vsTriangleDown } from '@deephaven/icons';
+import type { dh as DhType } from '@deephaven/jsapi-types';
+import { ObjectIcon } from './common';
+
+interface ConsoleObjectsMenuProps {
+  openObject: (object: DhType.ide.VariableDefinition) => void;
+  objects: readonly DhType.ide.VariableDefinition[];
+}
+
+function ConsoleObjectsMenu({
+  openObject,
+  objects,
+}: ConsoleObjectsMenuProps): ReactElement {
+  const [filterText, setFilterText] = useState('');
+  const searchRef = useRef<SearchInput>(null);
+
+  const actions: DropdownActions = useMemo(() => {
+    if (objects.length === 0) {
+      return [];
+    }
+    const searchActions = {
+      menuElement: (
+        <SearchInput
+          value={filterText}
+          onChange={e => {
+            setFilterText(e.target.value);
+          }}
+          className="console-menu"
+          ref={searchRef}
+        />
+      ),
+    };
+    const filteredObjects = filterText
+      ? objects.filter(
+          ({ title }: { title?: string }) =>
+            title != null &&
+            title.toLowerCase().indexOf(filterText.toLowerCase()) >= 0
+        )
+      : objects;
+    const objectActions = filteredObjects.map(object => ({
+      title: object.title,
+      action: () => {
+        openObject(object);
+      },
+      icon: <ObjectIcon type={object.type} />,
+    }));
+    return [searchActions, ...objectActions];
+  }, [objects, filterText, openObject]);
+
+  return (
+    <Button
+      aria-label="Objects"
+      kind="ghost"
+      disabled={objects.length === 0}
+      onClick={() => {
+        // no-op: click is handled in `DropdownMenu`
+      }}
+      tooltip={objects.length === 0 ? 'No objects available' : 'Objects'}
+      icon={
+        <div className="fa-md fa-layers">
+          <FontAwesomeIcon
+            mask={vsGraph}
+            icon={vsTriangleDown}
+            transform="right-5 down-5"
+          />
+          <FontAwesomeIcon icon={vsTriangleDown} transform="right-8 down-6" />
+        </div>
+      }
+    >
+      <DropdownMenu
+        actions={actions}
+        onMenuOpened={() => searchRef.current?.focus()}
+        onMenuClosed={() => setFilterText('')}
```

Comment:
I wonder if we should leave the filter and just have the search select the search text when it's reopened. I guess if this is the behavior it has always been and nobody has complained then it's inconsequential. I was just thinking if I'm searching maybe I wanted to open several matching the same search or something

Reply:
Yeah, could have it on a timer to reset it to blank after a few seconds. I could have sworn we had that logic on the query widget list on Enterprise, but I can't find it.

I'll just have it select text on open, that's how the QueryWidgetList behaves on Enterprise now.